internal/services/managedapplications/managed_application_resource.go

```diff
@@ -204,97 +202,98 @@ func resourceManagedApplicationRead(d *pluginsdk.ResourceData, meta interface{})
-	d.Set("location", location.NormalizeNilable(resp.Location))
-	d.Set("kind", resp.Kind)
-	if err := d.Set("plan", flattenManagedApplicationPlan(resp.Plan)); err != nil {
-		return fmt.Errorf("setting `plan`: %+v", err)
-	}
-	if props := resp.ApplicationProperties; props != nil {
-		id, err := resourcesParse.ResourceGroupIDInsensitively(*props.ManagedResourceGroupID)
+	if m := resp.Model; m != nil {
+		p := m.Properties
+
+		d.Set("name", m.Name)
+		d.Set("resource_group_name", id.ResourceGroupName)
+		d.Set("location", location.NormalizeNilable(m.Location))
+		d.Set("kind", m.Kind)
+		if err := d.Set("plan", flattenManagedApplicationPlan(m.Plan)); err != nil {
+			return fmt.Errorf("setting `plan`: %+v", err)
+		}
+
+		id, err := resourcesParse.ResourceGroupIDInsensitively(*p.ManagedResourceGroupId)
 		if err != nil {
 			return err
 		}
 
 		d.Set("managed_resource_group_name", id.ResourceGroup)
-		d.Set("application_definition_id", props.ApplicationDefinitionID)
+		d.Set("application_definition_id", p.ApplicationDefinitionId)
 
-		parameterValues, err := flattenManagedApplicationParameterValuesValueToString(props.Parameters)
+		parameterValues, err := flattenManagedApplicationParameterValuesValueToString(p.Parameters)
 		if err != nil {
 			return fmt.Errorf("serializing JSON from `parameter_values`: %+v", err)
 		}
 		d.Set("parameter_values", parameterValues)
 
-		parameters, err := flattenManagedApplicationParametersOrOutputs(props.Parameters)
+		parameters, err := flattenManagedApplicationParametersOrOutputs(p.Parameters)
 		if err != nil {
 			return err
 		}
 		if err = d.Set("parameters", parameters); err != nil {
 			return err
 		}
 
-		outputs, err := flattenManagedApplicationParametersOrOutputs(props.Outputs)
+		outputs, err := flattenManagedApplicationParametersOrOutputs(p.Outputs)
 		if err != nil {
 			return err
 		}
 		if err = d.Set("outputs", outputs); err != nil {
 			return err
 		}
+
+		return tags.FlattenAndSet(d, m.Tags)
 	}
 
-	return tags.FlattenAndSet(d, resp.Tags)
+	return nil
 }
 
 func resourceManagedApplicationDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).ManagedApplication.ApplicationClient
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationID(d.Id())
+	id, err := applications.ParseApplicationID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	err = client.DeleteThenPoll(ctx, *id)
 	if err != nil {
-		return fmt.Errorf("failed to delete Managed Application %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
-	}
-
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("failed to wait for deleting Managed Application (Managed Application Name %q / Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		return fmt.Errorf("failed to delete Managed Application %s: %+v", id, err)
```

Comment:
_pedantic_ but for consistency:

```suggestion
		return fmt.Errorf("deleting %s: %+v", *id, err)
```